### PR TITLE
backend/feat: frfs config table

### PR DESCRIPTION
**Pull request by chakri642**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## How did you test it?

- tested locally
Screenshot 2024-04-27 at 3 27 01 PM

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml [hunk 1]
```diff
@@ -392,3 +393,26 @@ FRFSRecon:
...
+  tableName: frfs_config
+  derives: "Generic,Show"
+  fields:
+    merchantOperatingCityId: Id MerchantOperatingCity
+    merchantId: Id Merchant
+    bookingStartTime: UTCTime
+    bookingEndTime: UTCTime
+    routndTripTicketLimit: Int
+    oneWayTicketLimit: Int
+    city: City
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml:407]: There is `merchantOperatingCityId`, do you need `city`?

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml [hunk 2]
```diff
@@ -392,3 +393,26 @@ FRFSRecon:
...
+
+  queries:
+    findByCity:
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml:410]: You need to lookup based on `merchantOperatingCityId`

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/FRFSTicketService.hs [hunk 3]
```diff
@@ -487,3 +488,15 @@ getAbsoluteValue mbRefundAmount = case mbRefundAmount of
   Just rfValue -> do
     let HighPrecMoney value = rfValue
     Just (HighPrecMoney $ abs value)
+
+getFrfsConfig :: (Kernel.Prelude.Maybe (Kernel.Types.Id.Id Domain.Types.Person.Person), Kernel.Types.Id.Id Domain.Types.Merchant.Merchant) -> API.Types.UI.FRFSTicketService.FRFSConfigAPIReq -> Environment.Flow API.Types.UI.FRFSTicketService.FRFSConfigAPIRes
+getFrfsConfig (_, _) FRFSConfigAPIReq {..} = do
+  frfsconfig <- QFRFSConfig.findByCity city >>= fromMaybeM (InvalidRequest "FRFS Config not found")
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/FRFSTicketService.hs:494]: This should be served from Cache and not DB. Also for fallback use replica DB

**Review by NikithShetty — CHANGES_REQUESTED** [review]

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml (hunk 2 above)
Comment (chakri642) [reply Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml:410]: yea i thought to do by that, but in frontend we don't get the `merchantOperatingCityId` at the place where we call our api, so we made based on `city`

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml (hunk 1 above)
Comment (chakri642) [reply Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml:407]: yes for frontend we are fetching the data using city

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/FRFSTicketService.hs [hunk 4]
```diff
@@ -487,3 +488,20 @@ getAbsoluteValue mbRefundAmount = case mbRefundAmount of
   Just rfValue -> do
     let HighPrecMoney value = rfValue
     Just (HighPrecMoney $ abs value)
+
+getFrfsConfig :: (Kernel.Prelude.Maybe (Kernel.Types.Id.Id Domain.Types.Person.Person), Kernel.Types.Id.Id Domain.Types.Merchant.Merchant) -> Context.City -> Environment.Flow API.Types.UI.FRFSTicketService.FRFSConfigAPIRes
+getFrfsConfig (_, merchantId) opCity = do
+  merchantOpCity <- CQMOC.findByMerchantIdAndCity merchantId opCity >>= fromMaybeM (MerchantOperatingCityNotFound $ "merchant-Id-" <> merchantId.getId <> " ,city: " <> show opCity)
+  frfsconfig <- B.runInReplica $ CQFRFSConfig.findByMerchantOperatingCityId merchantOpCity.id >>= fromMaybeM (InvalidRequest "FRFS Config not found")
+  let config =
+        FRFSTicketService.FRFSConfigAPIRes
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/FRFSTicketService.hs:497]: Use `{..}` to simplify

**Review by NikithShetty — CHANGES_REQUESTED** [review]

**Review by NikithShetty — APPROVED** [review]

**Pull request merged**